examples/simple/anariDistributedTutorial.cpp

```diff
@@ -0,0 +1,181 @@
+{
+  (void)userData;
+  (void)device;
+  (void)source;
+  (void)sourceType;
+  (void)code;
+  if (severity == ANARI_SEVERITY_FATAL_ERROR) {
+    fprintf(stderr, "[FATAL] %s\n", message);
+  } else if (severity == ANARI_SEVERITY_ERROR) {
+    fprintf(stderr, "[ERROR] %s\n", message);
+  } else if (severity == ANARI_SEVERITY_WARNING) {
+    fprintf(stderr, "[WARN ] %s\n", message);
+  } else if (severity == ANARI_SEVERITY_PERFORMANCE_WARNING) {
+    fprintf(stderr, "[PERF ] %s\n", message);
+  } else if (severity == ANARI_SEVERITY_INFO) {
+    fprintf(stderr, "[INFO ] %s\n", message);
+  } else if (severity == ANARI_SEVERITY_DEBUG) {
+    fprintf(stderr, "[DEBUG] %s\n", message);
+  }
+}
+
+static void onFrameCompletion(const void *, anari::Device d, anari::Frame f)
+{
+  printf("anari::Device(%p) finished rendering anari::Frame(%p)!\n", d, f);
+}
+
+int main(int argc, char **argv)
+{
+  int mpiThreadCapability = 0;
+  MPI_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &mpiThreadCapability);
+  if (mpiThreadCapability != MPI_THREAD_MULTIPLE
+      && mpiThreadCapability != MPI_THREAD_SERIALIZED) {
+    fprintf(stderr,
+        "MPI runtime must support either thread multiple or thread serialized.\n");
+    return 1;
+  }
+
+  int mpiRank = 0;
+  int mpiWorldSize = 0;
+  MPI_Comm_rank(MPI_COMM_WORLD, &mpiRank);
+  MPI_Comm_size(MPI_COMM_WORLD, &mpiWorldSize);
+
+  // image size
+  uvec2 imgSize = {1024 /*width*/, 768 /*height*/};
+
+  // camera
+  vec3 cam_pos{(mpiWorldSize + 1.f) / 2.f, 0.5f, -mpiWorldSize * 0.5f};
+  vec3 cam_up{0.f, 1.f, 0.f};
+  vec3 cam_view{0.f, 0.f, 1.f};
+
+  // all ranks specify the same rendering parameters, with the exception of
+  // the data to be rendered, which is distributed among the ranks
+  // triangle mesh data
+  vec3 vertex[] = {{(float)mpiRank, 0.0f, 3.5f},
+      {(float)mpiRank, 1.0f, 3.0f},
+      {mpiRank + 1.f, 0.0f, 3.0f},
+      {mpiRank + 1.f, 1.0f, 2.5f}};
+  vec4 color[] = {{0.0f, 0.0f, (mpiRank + 1.f) / mpiWorldSize, 1.0f},
+      {0.0f, 0.0f, (mpiRank + 1.f) / mpiWorldSize, 1.0f},
+      {0.0f, 0.0f, (mpiRank + 1.f) / mpiWorldSize, 1.0f},
+      {0.0f, 0.0f, (mpiRank + 1.f) / mpiWorldSize, 1.0f}};
+  uvec3 index[] = {{0, 1, 2}, {1, 2, 3}};
+
+  anari::Library lib = anari::loadLibrary("environment", statusFunc);
+
+  anari::Extensions extensions =
+      anari::extension::getDeviceExtensionStruct(lib, "default");
+
+  if (!extensions.ANARI_KHR_GEOMETRY_TRIANGLE)
+    printf("WARNING: device doesn't support ANARI_KHR_GEOMETRY_TRIANGLE\n");
+  if (!extensions.ANARI_KHR_CAMERA_PERSPECTIVE)
+    printf("WARNING: device doesn't support ANARI_KHR_CAMERA_PERSPECTIVE\n");
+  if (!extensions.ANARI_KHR_MATERIAL_MATTE)
+    printf("WARNING: device doesn't support ANARI_KHR_MATERIAL_MATTE\n");
+  if (!extensions.ANARI_KHR_FRAME_COMPLETION_CALLBACK) {
+    printf(
+        "INFO: device doesn't support ANARI_KHR_FRAME_COMPLETION_CALLBACK\n");
+  }
+
+  anari::Device d = anari::newDevice(lib, "distributed");
```

Comment:
This likely needs some discussion: do we want to define some convention / names to differentiate between a default device and a distributed one?
The (better?) alternative is to define an extension and search for devices supporting it, see also https://github.com/KhronosGroup/ANARI-Docs/issues/131